lib/grpc/server/router/template.ex

```diff
@@ -0,0 +1,99 @@
+defmodule GRPC.Server.Router.Template do
+  @moduledoc false
+  # https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#google.api.HttpRule
+  # Template = "/" Segments [ Verb ] ;
+  # Segments = Segment { "/" Segment } ;
+  # Segment  = "*" | "**" | LITERAL | Variable ;
+  # Variable = "{" FieldPath [ "=" Segments ] "}" ;
+  # FieldPath = IDENT { "." IDENT } ;
+  # Verb     = ":" LITERAL ;
+  @type segment_match :: String.t() | {atom(), [segment_match]}
+  @type matchers :: [segment_match]
+
+  @spec tokenize(binary(), [tuple()]) :: [tuple()]
+  def tokenize(path, tokens \\ [])
+
+  def tokenize(<<>>, tokens) do
+    Enum.reverse(tokens)
+  end
+
+  def tokenize(segments, tokens) do
+    {token, rest} = do_tokenize(segments, <<>>)
+    tokenize(rest, [token | tokens])
+  end
+
+  @terminals [?/, ?{, ?}, ?=, ?*]
+  defp do_tokenize(<<h, t::binary>>, <<>>) when h in @terminals do
+    {{List.to_atom([h]), []}, t}
+  end
```

Comment:
```suggestion
  for h <- [?/, ?{, ?}, ?=, ?*] do
    h_atom = List.to_atom([h])
    defp do_tokenize(<<unquote(h), t::binary>>, <<>>) do
      {{unquote(h_atom), []}, t}
    end
  end
```

mostly so that we avoid a blind `to_atom` in runtime. Who knows if `h in @terminals` ends up being removed in the future